nih-sparc/sparc-app
# oSPARC viewers

# Description

Adds the possibility for the users to open files in oSPARC.
* New option in file actions dropdown in the Files tab (dataset detail)
* Option opens a new modal window to select the viewer
![Peek 2021-03-03 12-24](https://user-images.githubusercontent.com/4764217/109798846-87d86880-7c1b-11eb-84f0-2540ec17e6ce.gif)
* First option selected by default
* Link to more information about oSPARC


## Type of change

New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

There are no automated tests for this feature. Manual testing has been done.
## How to test
* Needs https://github.com/nih-sparc/sparc-api/pull/34 running
* Go to a dataset detail page and open the files tab
* Go to one file and open the options (button on the right)
* If the file is viewable in oSPARC, the corresponding option will appear
* Clicking on the option opens a modal window to select the file
* A new window opens and the file is displayed within the oSPARC platform
* Repeat the operation for some other files
* You can check the available file viewers in https://osparc.io/v0/viewers


# Checklist:
